### ISSUE TITLE
Unable to read the response stream from an exception after using GetStreamAsync() with HttpCompletionOption.ResponseHeadersRead

### ISSUE THREAD

**Issue by cremor** [tmenier/Flurl#663]
To emulate the default behavior of `HttpClient` I always pass `HttpCompletionOption.ResponseHeadersRead` to`GetStreamAsync()` (see #630). But I noticed that this has a bad effect for error handling. If the HTTP call returns an HTTP error code and I then try to get the error response content with `FlurlHttpException.GetResponseStringAsync()` the return value is always null.

But if I revert the `GetStreamAsync()` call to not specify `HttpCompletionOption.ResponseHeadersRead` then the call to `FlurlHttpException.GetResponseStringAsync()` returns the correct response body.

Is there a way to get both?

Example code:

```cs
public async Task<Stream> DownloadFileAsync(int id) {
   try {
      return await _flurlClient.Request("download", id)
         .GetStreamAsync(completionOption: HttpCompletionOption.ResponseHeadersRead);
   } catch (FlurlHttpException ex) {
      var errorResponse = await ex.GetResponseJsonAsync<ErrorResponseType>();
      // OR
      var errorResponse = await ex.GetResponseStringAsync();
      // Do something with errorResponse
   }
}
```

**Comment by tmenier** [comment]
I can't repro this, but there's a very good chance that the fix for #571 also fixed this. That was released in January (after your reported this...sorry it took me a while to get to it). Can you confirm whether this is fixed with the latest version of Flurl.Http?

**Comment by cremor** [comment]
I'm not 100% sure (because I don't remember how I tested this exactly back then) but yes, it looks like this is fixed.
